zimbra/src/main/java/fr/openent/zimbra/security/RecallFilter.java

```diff
@@ -0,0 +1,22 @@
+package fr.openent.zimbra.security;
+
+import fr.wseduc.webutils.http.Binding;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpServerRequest;
+import org.entcore.common.http.filter.ResourcesProvider;
+import org.entcore.common.user.UserInfos;
+
+public class RecallFilter implements ResourcesProvider {
+
+    @Override
+    public void authorize(HttpServerRequest request, Binding binding, UserInfos user,
+                          Handler<Boolean> handler) {
+
+        if (WorkflowActionUtils.hasRight(user, WorkflowActions.RECALL_RIGHT.toString())) {
+            handler.handle(true);
+        } else {
+            handler.handle(false);
+        }
```

Comment:
Possibilité de faire en une ligne sans passer par if - else ?

```suggestion
        handler.handle(WorkflowActionUtils.hasRight(user, WorkflowActions.RECALL_RIGHT.toString()));
```